### ISSUE TITLE
Start container on boot

### ISSUE THREAD

**Issue by XenGi** [containers/quadlet#20]
It seems like I can't setup containers to start on boot. I tried the following:

```
$ systemctl enable mycontainer
Failed to enable unit: Unit /run/systemd/generator/mycontainer.service is transient or generated.
```

How would I achieve this with a container generated by quadlet?

My container file looks like this:

```ini
[Unit]
Description=My container
After=network.target

[Container]
Image=docker.io/mycontainer:latest

[Service]
Restart=always

[Install]
WantedBy=multi-user.target
```

AFAIK This has to be done by the generator, so in this case quadlet.

My service is defined in `/etc/container/systemd/mycontainer.container`. It generates the service file `/run/systemd/generated/mycontainer.service`correctly. It looks like this:

```ini
# Automatically generated by quadlet-generator
[Unit]
Description=My container
After=network.target
RequiresMountsFor=%t/containers
SourcePath=/etc/containers/systemd/mycontainer.container

[X-Container]
Image=docker.io/mycontainer:latest

[Service]
Restart=always
Environment=PODMAN_SYSTEMD_UNIT=%n
KillMode=mixed
ExecStartPre=-rm -f %t/%N.cid
ExecStopPost=-/usr/bin/podman rm -f -i --cidfile=%t/%N.cid
ExecStopPost=-rm -f %t/%N.cid
Delegate=yes
Type=notify
NotifyAccess=all
SyslogIdentifier=%N
ExecStart=/usr/bin/podman run --name=systemd-%N --cidfile=%t/%N.cid --replace --rm -d --log-driver journald --pull=never --runtime /usr/bin/crun --cgroups=split --init --sdnotify=conmon --security-opt=no-new-privileges --cap-drop=all --mount type=tmpfs,tmpfs-size=512M,destination=/tmp --uidmap 0:0:1 --uidmap 1:1879048192:899 --uidmap 901:1879049091:64637 --gidmap 0:0:1 --gidmap 1:1879048192:899 --gidmap 901:1879049091:64637 docker.io/mycontainer:latest

[Install]
WantedBy=multi-user.target
```

What it should also do is generate a symlink from `/run/systemd/generator/multi-user.target.wants/mycontainer.service` to `/run/systemd/generated/mycontainer.service`. That should be enough to get the container started at boot.

tl;dr:

When generating the `*.service` files from the `*.container` files quadlet should read the `Install` section and create corresponding symlinks in `/run/systemd/generator/...target.wants/` directories.

**Comment by XenGi** [comment]
AFAIK This has to be done by the generator, so in this case quadlet.

My service is defined in `/etc/container/systemd/mycontainer.container`. It generates the service file `/run/systemd/generated/mycontainer.service`. What it should also do is generate a symlink from `/run/systemd/generator/multi-user.target.wants/mycontainer.container` to `/run/systemd/generated/mycontainer.service`. That should be enough to get the container started at boot.

**Comment by b-m-f** [comment]
@XenGi As a workaround you could `@reboot` start the services inside a `crontab -e` until this is supported

**Comment by XenGi** [comment]
Yeah, but that just feels wrong. :)

**Comment by XenGi** [comment]
I would give it a try and work on a pull request for that feature but before I do that I would like to see some activity in the other issues.
I'm currently not sure if this is still maintained. :/

**Comment by alexlarsson** [comment]
Sorry for not being very attentative, been somewhat busy with other things.
However, this seems pretty important, I'll have a look at it.

**Comment by b-m-f** [comment]
No rush :) 

If you figure sth out and need help with coding @XenGi or me could surely give it a try.

**Comment by XenGi** [comment]
@b-m-f we could have a remote coding session if you like. My C is a bit rusty but creating some symlinks shouldn't be too hard.

**Comment by b-m-f** [comment]
Hey @XenGi ,

Im in Berlin and have some time off over christmas.

Gonna ping you on monday on the mail in your profile if that works for you!

**Comment by XenGi** [comment]
@b-m-f Sure. I'm in Berlin too. Will be with family over the holidays but I'm free from Monday on. Just drop me a mail then.

**Comment by alexlarsson** [comment]
I did some initial work on this today. I think it'll work, but I need to test it more and add some tests to the testsuite.

**Comment by XenGi** [comment]
@b-m-f and I will meet tomorrow to work on it further. So we could take your approach and finalize it.

**Comment by XenGi** [comment]
So we had a look at your code, compiled it and went through some test cases on my server.
It seems that this works pretty well.

The `[Install]` section with the `WantedBy` and `RequiredBy` keys is picked up correctly and symlinks are generated in the right places. `Wanted` and `Required` from the `[Unit]` section is automatically handled by systemd so I guess we don't need to fiddle with that.
Dependency resolution is also done by systemd so I guess we don't need to do any checks if service files exist etc. No need to duplicate functionality that is already there.
After a reboot everything is generated correctly and services bootup. We tested the effects of `WantedBy` and `RequiredBy` only indirectly by checked if the symlinks get created but that worked, so I guess it will work after a reboot.

All my services now start on boot so I'm quite happy.
Thx @alexlarsson for your work. Great job!
